frontend/hooks/useLikesCookie.js

```diff
@@ -0,0 +1,38 @@
+import { useCookies } from "react-cookie";
+
+export default () => {
+    const [cookies, setCookie] = useCookies(["likes"]);
+
+    var inTenYears = new Date();
+    inTenYears.setFullYear(inTenYears.getFullYear() + 10);
+
+    let savedLikes = cookies.likes ? cookies.likes : [];
```

Comment:
Better store the likes in an object... 
you could store the id of the idea in the object as key and just the value `true` as object-value. 
Then your access becomes more perfomant.

`const likesPerIdea = {};` 

and in the function: 
```
const toggleLike = (id) {
  if (id in likesPerIdea) {
    likesPerIdea[id]  = true;
  } else {
    delete likesPerIdea[id];
  }
}
```

in isLike you can just use now: 

```
const isLiked = (id) =>
  return (id in likesPerIdea)
};
```

Reply:
Here is the benchmark of objects vs arrays: https://jsben.ch/Y9jDP